Panic when submitting non-existing files to Load()

Hi,

I'm submitting a slice of three files to `ini.Load()`:

```go
cfg, err := ini.Load("/etc/securepass.conf", "/usr/local/etc/securepass.conf", "securepass.conf")
```

Only the latter exists and has the correct permissions set. Likewise the Python `ConfigParser` module, it would be great if the function could read the configuration from only those accessible or return an empty configuration if no existing files were provided.

```console
$ ~/go/src/github.com/garlsecurity/go-securepass$ go run spctl/main.go 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4811b2]

goroutine 1 [running]:
gopkg.in/ini%2ev1.(*File).GetSection(0x0, 0x7fa130, 0x7, 0x0, 0x0, 0x0)
	/home/SMARTODDS/tregliaa/go/src/gopkg.in/ini.v1/ini.go:976 +0x72
main.loadConfiguration()
	/home/SMARTODDS/tregliaa/go/src/github.com/garlsecurity/go-securepass/spctl/main.go:50 +0xc2
main.main()
	/home/SMARTODDS/tregliaa/go/src/github.com/garlsecurity/go-securepass/spctl/main.go:60 +0x1c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1
exit status 2
```

Thanks for considering.

--- a/ini.go
+++ b/ini.go
@@ -123,16 +123,20 @@ type File struct {
 	// To keep data in order.
 	sectionList []string
 
+	// whether the parser should check for files existence first
+	looseMode bool
+
 	NameMapper
 }
 
 // newFile initializes File object with given data sources.
-func newFile(dataSources []dataSource) *File {
+func newFile(dataSources []dataSource, looseMode bool) *File {
 	return &File{
 		BlockMode:   true,
 		dataSources: dataSources,
 		sections:    make(map[string]*Section),
 		sectionList: make([]string, 0, 10),
+		looseMode:   looseMode,
 	}
 }
 
@@ -150,6 +154,17 @@ func parseDataSource(source interface{}) (dataSource, error) {
 // Load loads and parses from INI data sources.
 // Arguments can be mixed of file name with string type, or raw data in []byte.
 func Load(source interface{}, others ...interface{}) (_ *File, err error) {
+	return loadSources(false, source, others...)
+}
+
+// Load loads and parses from INI data sources and ignores errors.
+// Arguments can be mixed of file name with string type, or raw data in []byte.
+// If a file listed in data sources cannot be opened, that file will be ignored.
+func LooseLoad(source interface{}, others ...interface{}) (_ *File, err error) {
+	return loadSources(true, source, others...)
+}
+
+func loadSources(looseMode bool, source interface{}, others ...interface{}) (_ *File, err error) {
 	sources := make([]dataSource, len(others)+1)
 	sources[0], err = parseDataSource(source)
 	if err != nil {
@@ -161,7 +176,7 @@ func Load(source interface{}, others ...interface{}) (_ *File, err error) {
 			return nil, err
 		}
 	}
-	f := newFile(sources)
+	f := newFile(sources, looseMode)
 	if err = f.Reload(); err != nil {
 		return nil, err
 	}
@@ -284,7 +299,7 @@ func (f *File) reload(s dataSource) error {
 // Reload reloads and parses all data sources.
 func (f *File) Reload() (err error) {
 	for _, s := range f.dataSources {
-		if err = f.reload(s); err != nil {
+		if err = f.reload(s); err != nil && !f.looseMode {
 			return err
 		}
 	}

--- a/ini_test.go
+++ b/ini_test.go
@@ -116,6 +116,10 @@ func Test_Load(t *testing.T) {
 			cfg, err := Load([]byte(_CONF_DATA), "testdata/conf.ini")
 			So(err, ShouldBeNil)
 			So(cfg, ShouldNotBeNil)
+
+			f, err := Load([]byte(_CONF_DATA), "testdata/404.ini")
+			So(err, ShouldNotBeNil)
+			So(f, ShouldBeNil)
 		})
 	})
 
@@ -166,6 +170,33 @@ func Test_Load(t *testing.T) {
 			So(err, ShouldNotBeNil)
 		})
 	})
+}
+
+func Test_LooseLoad(t *testing.T) {
+	Convey("LooseLoad from data sources", t, func() {
+
+		Convey("LooseLoad from invalid data sources", func() {
+			_, err := LooseLoad(_CONF_DATA)
+			So(err, ShouldBeNil)
+
+			f, err := LooseLoad("testdata/404.ini")
+			So(err, ShouldBeNil)
+			So(f, ShouldNotBeNil)
+
+			_, err = LooseLoad(1)
+			So(err, ShouldNotBeNil)
+
+			_, err = LooseLoad([]byte(""), 1)
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("LooseLoad with multiple data sources", func() {
+			cfg, err := LooseLoad([]byte(_CONF_DATA), "testdata/404.ini")
+			So(err, ShouldBeNil)
+			So(cfg, ShouldNotBeNil)
+		})
+	})
+
 }
 
 func Test_File_Append(t *testing.T) {

